src/amplitude-client.js

```diff
@@ -212,6 +212,31 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
     if (type(opt_callback) === 'function') {
       opt_callback(this);
     }
+
+    const onExitPage = this.options.onExitPage;
+    if (type(onExitPage) === 'function') {
+      if (!this.pageHandlersAdded) {
+        this.pageHandlersAdded = true;
+
+        const handleVisibilityChange = () => {
+          const prevTransport = this.options.transport;
+          this.setTransport(Constants.TRANSPORT_BEACON);
+          onExitPage();
```

Comment:
thoughts on hooking the current object with `onExitPage(this)` allowing users to access the client earlier?

Reply:
It might make it easier for people using the snippet to use the full client obj (incl. "private" methods and the full list of options post-init to construct event properties) but you are right that everything publicly supported is available. 

I don't feel too strongly about this though so we can not do this and see what people actually want as arguments to this option!